src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -139,6 +149,8 @@ public async Task ConnectGetsCanceledByDispose(string addressString, bool useDns
             await RetryHelper.ExecuteAsync(async () =>
             {
                 var client = new Socket(address.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+                using SafeSocketHandle? owner = ReplaceWithNonOwning(ref client, owning);
```

Comment:
We do not keep the reference of the original `Socket` instance after this line. Couldn't this lead to an unexpected eager finalization of that instance while `client` and `owner` are still alive?

Otherwise it will be just an ignored failing `close()` syscall in the finalizer, right?

Reply:
If we're replacing `client` (when `owning` is `true`) the original `Socket` gets:
```
            socket.SafeHandle.SetHandleAsInvalid();
```
so it will no longer `close` the platform socket.